compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -812,8 +816,17 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
             }
         }
 
-        if (targetTag == TypeTags.READONLY &&
-                (isInherentlyImmutableType(source) || Symbols.isFlagOn(source.flags, Flags.READONLY))) {
+        if (targetTag == TypeTags.READONLY) {
+            if ((isInherentlyImmutableType(source) || Symbols.isFlagOn(source.flags, Flags.READONLY))) {
+                return true;
+            }
+            // TODO: check this will make `any|error is readonly` always true
```

Comment:
Is this comment still valid? The following doesn't give an error and compiles, as expected.

```ballerina
public function main() {
    any|error x = 1;
    boolean b = x is readonly;
}
```

Reply:
Yes no need of this now.